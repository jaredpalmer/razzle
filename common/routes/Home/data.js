const data = [
  {
    resource: 'Node.js',
    link: 'https://nodejs.org/en/',
    description: "Node.jsÂ® is a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node.js' package ecosystem, npm, is the largest ecosystem of open source libraries in the world."
  },
  {
    resource: 'Express',
    link: 'https://github.com/expressjs/express',
    description: 'Fast, unopinionated, minimalist web framework for Node.js'
  },
  {
    resource: 'React',
    link: 'https://github.com/facebook/react',
    description: 'A declarative, efficient, and flexible JavaScript library for building user interfaces.'
  },
  {
    resource: 'Redux',
    link: 'https://github.com/reactjs/redux',
    description: 'Predictable state container for JavaScript apps'
  },
  {
    resource: 'React Router 2.0',
    link: 'https://github.com/reactjs/react-router',
    description: 'A complete routing solution for React.js'
  },
  { resource: 'Aphrodite for CSS',
    link: 'https://github.com/Khan/aphrodite',
    description: "Support for colocating your styles with your JavaScript component. It's inline styles, but they work!"
  },
  {
    resource: 'React Helmet',
    link: 'https://github.com/nfl/react-helmet',
    description: 'A document head manager for React'
  },
  { resource: 'Redial for data fetching',
    link: 'https://github.com/markdalgleish/redial',
    description: 'Universal data fetching and route lifecycle management for React'
  },
  { resource: 'Babel 6',
    link: 'https://github.com/babel/babel',
    description: 'Babel is a compiler for writing next generation JavaScript.'
  },
  {
    resource: 'Webpack',
    link: 'https://github.com/webpack/webpack',
    description: 'Webpack is a bundler for modules. The main purpose is to bundle JavaScript files for usage in a browser, yet it is also capable of transforming, bundling, or packaging just about any resource or asset.'
  }
]

export default data
