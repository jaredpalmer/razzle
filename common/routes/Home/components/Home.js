import React from 'react'

import { StyleSheet, css } from 'aphrodite'
import data from '../data'

// This is a static page. It uses an array to hold data about the resources
// and maintain DRY
const Home = (props) => (
  <div>

    <h2 className={css(styles.header)}>About</h2>
    <p className={css(styles.lead)}>
      This is an example react application (master-detail feed) with isomorphic rendering, async react-router routes, async redux reducers, async data fetching, and code-splitting.
    </p>
    <h2 className={css(styles.header)}>Motivation</h2>
    <p className={css(styles.lead)}>
      The file size of isomorphic React apps can quickly get out of hand. Many isomorphic starter kits look awesome to begin with but yield a several megabyte javascript
      file for the client to download. This project aims to demonstrate some possible solutions.
    </p>
    <h2 className={css(styles.header)}>Under the Hood</h2>
    <ul className={css(styles.list)}>
      {data.map((item, i) => (
        <li key={i}>
          <h3><a className={css(styles.link)} href={item.link} target='_blank'>{item.resource}</a></h3>
          <p className={css(styles.body)}>{item.description}</p>
        </li>
       ))}
    </ul>
  </div>
)

const styles = StyleSheet.create({
  header: {
    fontSize: 28,
    lineHeight: '1.2',
    margin: '0 0 1.5rem'
  },
  lead: {
    fontSize: 18,
    lineHeight: '1.5',
    margin: '0 0 1.5rem',
    color: '#555'
  },
  body: {
    fontSize: '1rem',
    lineHeight: '1.5',
    margin: '0 0 1.5rem',
    color: '#555'
  },
  list: {
    fontSize: '1rem',
    listStyle: 'none',
    padding: 0
  },
  link: {
    display: 'block',
    fontSize: '1.25rem',
    margin: '0 0 .5rem',
    lineHeight: '1.5',
    fontWeight: 'bold',
    color: '#08c',
    opacity: 1,
    transition: '.2s opacity ease',
    textDecoration: 'none',
    ':hover': {
      opacity: 0.5,
      textDecoration: 'none'
    }
  }
})

export default Home
