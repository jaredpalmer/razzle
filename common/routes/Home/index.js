import Home from './components/Home'

export default Home
