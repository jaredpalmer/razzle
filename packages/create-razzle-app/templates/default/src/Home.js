import React, { Component } from 'react';
import logo from './react.svg';
import styles from './Home.module.css';
import './Home.css';

class Home extends Component {
  render() {
    return (
      <div className={styles.home}>
        <div className={styles.header}>
          <img src={logo} className="Home-logo" alt="logo" />
          <h2>Welcome to Razzle</h2>
        </div>
        <p className={styles.intro}>
          To get started, edit <code>src/App.js</code> or{' '}
          <code>src/Home.js</code> and save to reload.
        </p>
        <ul className={styles.resources}>
          <li>
            <a href="https://github.com/jaredpalmer/razzle">Docs</a>
          </li>
          <li>
            <a href="https://github.com/jaredpalmer/razzle/issues">Issues</a>
          </li>
          <li>
            <a href="https://palmer.chat">Community Slack</a>
          </li>
        </ul>
      </div>
    );
  }
}

export default Home;
