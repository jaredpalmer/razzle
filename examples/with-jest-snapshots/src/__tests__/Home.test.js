import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';

it('renders Home correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
