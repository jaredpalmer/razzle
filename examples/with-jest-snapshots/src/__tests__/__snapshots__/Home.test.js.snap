// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<div>
  Hello World!
</div>
`;
